docs/architecture/modernize-desktop/example-migration-to-net-core.md

```diff
@@ -0,0 +1,365 @@
+Registry, ACLs, WCF you have to include a reference to the
+Microsoft.Windows.Compatibility package to add these Windows-specific APIs. They
+work on .NET Core but aren't included as they aren't cross-platform.
+
+There is a tool called API Analyzer
+(<https://docs.microsoft.com/en-us/dotnet/standard/analyzers/api-analyzer>) that
+help you identify not compatible APIs as long as you develop your code.
+
+#### Use \#if defines
+
+If you need different execution paths when targeting .NET Framework and .NET
+Core, you should use compilation constants and code some \#if statements to keep
+the same code base for both targets.
+
+#### Technologies not available on .NET Core
+
+Some technologies aren't available on .NET Core like:
+
+* AppDomains
+* Remoting
+* Code Access Security
+* WCF Server
+* Windows Workflow
+
+Therefore, it is time to look for a replacement for those.
+
+#### Regenerate autogenerated clients
+
+If your application uses some autogenerated code like, for example, a WCF
+client, you may need to regenerate this code to target .NET Core. Sometimes you
+can find some missing references since it may not be included as default as part
+of the basic .Net Core assemblies. Using a tool like <https://apisof.net/> you
+can easily locate the assembly the missing reference lives in and add it from
+NuGet.
+
+#### Rolling back package versions
+
+As a general rule, we have stated that you better update every single package
+version to be compatible with .NET Core. However, you can find that targeting an
+updated and compatible version of an assembly just does not pay off. If the cost
+of change is not acceptable, you can consider rolling back package versions
+keeping the ones you use on .NET Framework. Although they may not be targeting
+.NET Core, they should work well unless they call some unsupported APIs.
+
+### Run and Test
+
+Once you have your application building with no errors, you can start the last
+step of the migration by testing every functionality.
+
+In this final step, you can find a variety of issues depending on the complexity
+of your application and the dependencies and APIs you are using.
+
+For example, in the case you have some usage of configuration files (app.config)
+you may find some errors at run time like Configuration Sections not present.
+Using Microsoft.Extensions.Configuration should do the fix.
+
+Another reason for errors is the use of BeginInvoke and EndInvoke because they
+are supported on .NET Core. The reason they aren't supported on .NET Core is
+they have a dependency on Remoting, which does not exist on .NET Core. To solve
+this issue, try to use `await` when available or `Task.Run`.
+
+You can use compatibility analyzers to let you identify APIs and code patterns
+in your code that can potentially cause problems at run-time with .NET Core. You
+can go to <http://github.com/dotnet/platform-compat> and use the Roselyn
+analyzers.
+
```

Comment:
We should call it .NET API Analyzer since we productized it with this name. Compatibility Analizer is the old internal name. but when we went public, it is refereed as .NET API Analyzer.